active-record/src/main/java/com/iluwatar/active/record/MagicWand.java

```diff
@@ -0,0 +1,261 @@
+package com.iluwatar.active.record;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * "Every single wand is unique and will depend for its character on the particular tree and magical creature
+ * from which it derives its materials. Moreover, each wand, from the moment it finds its ideal owner, will begin
+ * to learn from and teach its human partner." @ Garrick Ollivander
+ * <p/>
+ * Created by Stephen Lazarionok.
+ */
+public class MagicWand {
+
+    private Long id;
+
+    private double lengthInches;
+
+    private WandWoodType wood;
+
+    private WandCoreType core;
+
+    public Long getId() {
+        return id;
+    }
+
+    public double getLengthInches() {
+        return lengthInches;
+    }
+
+    public WandWoodType getWood() {
+        return wood;
+    }
+
+    public WandCoreType getCore() {
+        return core;
+    }
+
+    public void setLengthInches(double lengthInches) {
+        this.lengthInches = lengthInches;
+    }
+
+    public void setWood(WandWoodType wood) {
+        this.wood = wood;
+    }
+
+    public void setCore(WandCoreType core) {
+        this.core = core;
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder("MagicWand(");
+        sb.append(getWood());
+        sb.append(", ");
+        sb.append(getCore());
+        sb.append(", ");
+        sb.append(getLengthInches());
+        sb.append(" inch(es))");
+        return sb.toString();
+    }
+
+    /**
+     * ***********************************
+     * ****** Data access methods
+     * ***********************************
+     */
+
+    private static final String SELECT_SQL = "select * from wand where id = ?";
+    private static final String DELETE_SQL = "delete from wand where id = ?";
+    private static final String UPDATE_SQL = "update wand set length_inches = ?, wood = ?, core = ? where id = ?";
+    private static final String CREATE_SQL = "insert into wand values(?, ?, ?, ?)";
+
+    /**
+     * Saves the instance to the DB.
+     * @return
+     */
+    public long save() {
+        validateToSave();
+        try {
+            final Connection connection = DB.getConnection();
+            final PreparedStatement ps = connection.prepareStatement(CREATE_SQL);
+
+            final long id = System.currentTimeMillis();
+            ps.setLong(1, id);
+            ps.setDouble(2, getLengthInches());
+            ps.setString(3, getWood().name());
+            ps.setString(4, getCore().name());
+
+            ps.execute();
+
+            ps.close();
```

Comment:
I am nitpicking here but it is not a good idea to close connections in `try` block, they must be in `finally` block. This does not matter in our small example. But I would prefer we don't violate best practices.